src/member-api.js

```diff
@@ -0,0 +1,44 @@
+import { TypedEmitter } from 'tiny-typed-emitter'
+import { MapeoRPC } from './rpc/index.js'
+
+export class MemberApi extends TypedEmitter {
+  #encryptionKeys
+  #getProjectInfo
+  #projectKey
+  #rpc
+
+  /**
+   * @param {Object} opts
+   * @param {import('./generated/keys.js').EncryptionKeys} opts.encryptionKeys
+   * @param {() => Promise<import('./generated/rpc.js').Invite_ProjectInfo>} opts.getProjectInfo
+   * @param {Buffer} opts.projectKey
+   * @param {MapeoRPC} opts.rpc
+   */
+  constructor({ encryptionKeys, getProjectInfo, projectKey, rpc }) {
+    super()
+    this.#encryptionKeys = encryptionKeys
+    this.#getProjectInfo = getProjectInfo
+    this.#projectKey = projectKey
+    this.#rpc = rpc
+  }
+
+  /**
+   * @param {string} deviceId
+   *
+   * @param {Object} opts
+   * @param {string} opts.role
```

Comment:
what is this and how should it be used with MapeoRPC?

Reply:
how does the device being invited know what role they will be assigned prior to responding? Shouldn't the call to `rpc.invite()` also accept some param about the role to send to the target peer?

My understanding is that from a UX perspective, someone with an invite should see something like "Bob has invited you to be _ROLE_ in _PROJECT NAME_"

Reply:
param name addressed via https://github.com/digidem/mapeo-core-next/pull/232/commits/88a3ef4cfeefc90adabf4ec6c38ab1d3fc952cd6

Reply:
role assignment addressed via https://github.com/digidem/mapeo-core-next/pull/232/commits/e26ff99d4f47f3a77cd002df75ef2b31d655e987